src/phases/move-effect-phase.ts

```diff
@@ -308,6 +312,19 @@ export class MoveEffectPhase extends HitCheckPhase {
     }
   }
 
+  /**
+   * G-Max Moves apply their effects to both opponents (even if the target faints) and also
+   * ignores substitutes
+   *
+   * TODO: G-Max move edge cases like G-max Snooze
+   */
+  private applyGMaxEffects(user: Pokemon, target: Pokemon, hitResult: HitResult, firstTarget: boolean) {
+    this.applyOnTargetEffects(user, target, hitResult, firstTarget);
+    if (target.getAlly()?.isActive(true)) {
+      this.applyOnTargetEffects(user, target.getAlly(), hitResult, firstTarget);
```

Comment:
```suggestion
      this.triggerMoveEffects(MoveEffectTrigger.POST_APPLY, user, target.getAlly(), firstTarget, false);
```

I've realized that `applyOnTargetEffects` applies too many effects (e.g., it also applies the user's post-attack abilities against the target's ally). We only want to be applying the G-Max move's move attrs.